src/algebra/polynomial/big_operators.lean

```diff
@@ -90,6 +90,15 @@ begin
   rw prod_eq_one, intros, apply h, assumption,
 end
 
+lemma coeff_zero_prod :
```

Comment:
For `mv_polynomial` there is a `ring_hom` called `constant_coeff`. We should add it for `polynomial` as well, in a later PR.

Reply:
At which point this lemma would become `:= constant_coeff.map_prod`?

Reply:
We are quite far from that I think. To write the cyclotomic polynomials as a product we need Moebius inversion formula (that we almost have, it is in some PR), but still `cyclotomic n R` is the product of rational functions, not of polynomials, so it is really more complicated then this proof I would say. Here I use the product formula for `X ^ n - 1` as product of `cyclotomic d R`, when `d` divides `n`. The annoying part is that I have to separate the factor `n` (easy using `proper_divisors`) but also the factor `1`, that gives a constant coefficient of `-1` matching with the `-1 of `X ^ n - 1`.

Reply:
@riccardobrasca I think Eric meant the little lemma that we are commenting on. Not the other results in this PR.
(Also Moebius inversion was merged this morning :tada:)

Reply:
Ah, yes! If we had `constant_coeff` as a `ring_home` we didn't need `coeff_zero_prod` at all.